Sources/SwiftParserDiagnostics/MissingNodesError.swift

```diff
@@ -290,6 +290,51 @@ public struct InsertTokenFixIt: ParserFixIt {
 // MARK: - Generate Error
 
 extension ParseDiagnosticsGenerator {
+  func handleMissingToken(_ node: TokenSyntax) {
+    guard let invalidToken = node.previousToken(viewMode: .all),
+          let unexpectedTokens = invalidToken.parent?.as(UnexpectedNodesSyntax.self),
+          unexpectedTokens.count == 1 else {
+      _ = handleMissingSyntax(node)
+      return
+    }
+
+    // The previous token is unexpected, assume that it was intended to be
+    // this token.
+
+    if node.tokenKind.isIdentifier {
+      let fixIts: [FixIt]
+      if invalidToken.tokenKind.isKeyword || invalidToken.tokenKind.isDollarIdentifier {
+        // TODO: Should the parser add the text with backticks to the missing
+        // node? Then this could just make missing/present.
+        fixIts = [FixIt(message: .wrapInBackticks, changes: [
+          .replace(
+            oldNode: Syntax(invalidToken),
+            newNode: Syntax(TokenSyntax.identifier("`\(invalidToken.text)`", leadingTrivia: invalidToken.leadingTrivia, trailingTrivia: invalidToken.trailingTrivia))
+          )
+        ])]
+      } else {
+        fixIts = []
+      }
+      addDiagnostic(
+        invalidToken,
+        InvalidIdentifierError(invalidIdentifier: invalidToken, missingIdentifier: node),
+        fixIts: fixIts,
+        handledNodes: [unexpectedTokens.id]
+      )
+    } else if node.tokenKind == .period && invalidToken.tokenKind == .period {
+      let fixIt =  FixIt(message: .removeExtraneousWhitespace, changes: [
```

Comment:
Do we need to check first whether the invalid token actually had more trivia than the missing one?

Reply:
I don't think so, this pattern should never happen except in the case of the extraneous whitespace right? Unless you made your own tree I guess but... do we care about that case?